### PR TITLE
All courses page

### PR DESCRIPTION
**Pull request by METrimble**
## Description
landing page now grabs information from the api to get the courses for the user and display them. 
Additional edits:
- Commented out portions of old frontend code we no longer need, but may want to reference
- Added new landing route to app.js, and commented out no longer needed routes
- Edited MainPage.js to contain reflect the new routes
- Edited CourseCard.js to omit (comment out) old elements we no longer want
- Added buttons in CourseCard.js which lead to the single course pages 

Please let me know of any changes I can make!

### Special Notes
Add course needed to be revaluated in placement 

### How has this been tested?
I ran the page using three different cases:
- Only teacher courses
- Only student courses
- Both 
- neither

This isn't extensive, however, if there's more testing that can be done please let me know. 

## Requirements
Fill out the checkboxes below, verifying that all the requirements have been satisfied:
- [X] Have I written new tests to show my code works?
- [X] Do all other tests still work?
- [X] Have I documented where necessary (technical docs, how to docs, etc.)?
- [X] Have I requested at least 1 reviewer?
- [X] Have I left comments so my work is easy to understand?
- [X] Have I looked over my code again to weed out any mistakes I may have made or details missed?
- [X] Does my work reflect all the needs of this given task?

### CODE REVIEW

**Review by gavenrobertson — APPROVED** [review]
Looks great! I think the implementation of the useState() looks good, this should be really easy to adjust into a redux implementation when the time comes.

File: .env.example [hunk 1]
```diff
@@ -1 +0,0 @@
-REACT_APP_API_URL='http://localhost:3001/'
```
Comment (ghost) [inline .env.example:1]: I'm a little confused why this was removed?

File: src/pages/InstructorStudentLanding.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                
+                //get the role from enrollment 
+
+            }
+            populateCourses()
+
+        }, [])
+
+        //student and teacher
+        if(studentCourses.length > 0 && !editToggle && instructorCourses.length > 0){
```
Comment (ghost) [inline src/pages/InstructorStudentLanding.js]: I'm a little confused by the conditionals here if I'm being completely honest. What's edit toggle? And then also, do you think there is a cleaner way to minimize the code that's being repeated in each of the blocks following the conditional? Say, embed the conditions in the JSX? I trust it works, just a matter of making it easier to read and understand (and shorter 😄 )

**Review by ghost — CHANGES_REQUESTED** [review]
Good stuff Elayne :) If you wouldn't mind adding the .env.example file back, then you have my approval! I also left a question for my own understanding of how things are working. Thanks!

File: src/pages/InstructorStudentLanding.js (hunk 2 above)
Comment (METrimble) [reply src/pages/InstructorStudentLanding.js]: I honestly have no idea what edit toggle is, I was trying to preserve the original logic while also adding student vs teacher functionality to one page lol. I'll mess with the edit toggle to actually see what it does. I'll look into embedded conditions  as well and how they work, I originally had it in two separate if's, but I could only return one thing so it never got to the other one.

File: .env.example (hunk 1 above)
Comment (METrimble) [reply .env.example:1]: Not sure either, I'll add it back

File: src/pages/InstructorStudentLanding.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { useState, useEffect } from 'react';
+import CourseCard from '../components/CourseCard'
+import { Button, Card } from "react-bootstrap"
+import courseData from "./data/courseData.json"
+import apiUtil from '../utils/apiUtil'
+
+function InstructorStudentLanding(props) {
```
Comment (ghost) [inline src/pages/InstructorStudentLanding.js:10]: I forgot to add @METrimble, would it just make sense to call this `Landing`?

**Pull request merged**